tls/extensions/s2n_client_psk.c

```diff
@@ -227,6 +227,69 @@ static S2N_RESULT s2n_select_psk_identity(struct s2n_connection *conn, struct s2
     return S2N_RESULT_OK;
 }
 
+/**
+ *= https://tools.ietf.org/html/rfc8446#section-4.2.10  
+ *# For PSKs provisioned via NewSessionTicket, a server MUST validate
+ *#  that the ticket age for the selected PSK identity (computed by
+ *#  subtracting ticket_age_add from PskIdentity.obfuscated_ticket_age
+ *#  modulo 2^32) is within a small tolerance of the time since the ticket
+ *#  was issued (see Section 8).
+ **/
+static S2N_RESULT s2n_validate_ticket_lifetime(struct s2n_connection *conn, uint32_t obfuscated_ticket_age, uint32_t ticket_age_add) 
+{
+    RESULT_ENSURE_REF(conn);
+
+    /* Subtract the ticket_age_add value from the ticket age in milliseconds. The resulting uint32_t value
+     * may wrap, resulting in the modulo 2^32 operation. */
+    uint32_t ticket_age_in_millis = obfuscated_ticket_age - ticket_age_add;
+    uint32_t session_lifetime_in_millis = conn->config->session_state_lifetime_in_nanos / ONE_MILLISEC_IN_NANOS;
+    RESULT_ENSURE_LTE(ticket_age_in_millis, session_lifetime_in_millis);
+
+    return S2N_RESULT_OK;
+}
+
+static S2N_RESULT s2n_select_resumption_psk(struct s2n_connection *conn, struct s2n_stuffer *wire_identities_in) {
```

Comment:
We could use the `struct s2n_offered_psk_list` for `wire_identities_in`.

Reply:
The s2n_offered_psk_list is for customers, so it doesn't let them access the obfuscated ticket age. I need access that value for resumption psks, so I have to use the wire_identites_in stuffer.

Reply:
The external psk has no ticket age. There is no use case for a customer accessing that field.